Ponca/src/Fitting/leastSquareLine.hpp

```diff
@@ -0,0 +1,66 @@
+
+
+template < class DataPoint, class _WFunctor, typename T>
+void
+LeastSquareLine<DataPoint, _WFunctor, T>::init(const VectorType& _evalPos)
+{
+    
+    // Setup primitive
+    Base::resetPrimitive();
+    Base::basisCenter() = _evalPos;
+
+    // Setup fitting internal values
+    m_sum        = Scalar(0.0);
+    m_cog         = VectorType::Zero();
+    m_cov         = MatrixType::Zero();
+}
+
+template < class DataPoint, class _WFunctor, typename T>
+bool
+LeastSquareLine<DataPoint, _WFunctor, T>::addNeighbor(const DataPoint& _nei)
+{
+    
+    VectorType q = _nei.pos() - Base::basisCenter();
+
+    m_cog  +=  q;
+    m_sum++;
+    m_cov  +=  q * q.transpose();
+    ++(Base::m_nbNeighbors);
+
+    return true;  
+}
+
+template < class DataPoint, class _WFunctor, typename T>
+FIT_RESULT
+LeastSquareLine<DataPoint, _WFunctor, T>::finalize ()
+{
+    
+  
+    // handle specific configurations
+    // With less than 3 neighbors the fitting is undefined
+    if(m_sum == Scalar(0.) || Base::m_nbNeighbors < 3)
```

Comment:
In the case of a line, 2 should be enough, right ?
```suggestion
    // With less than 2 neighbors the fitting is undefined
    if(m_sum == Scalar(0.) || Base::m_nbNeighbors < 2)
```